fitbenchmarking/systests/test_regression.py

```diff
@@ -240,12 +154,44 @@ def diff_result(actual, expected):
     return diff, msg
 
 
-def setup_options(multi_fit=False) -> Options:
+def compare_results(problem_sub_directory: str, result_filename: str) -> list:
+    """
+    Compares the expected benchmark results with the actual results,
+    and returns the lines which differ between expected and actual
+    along with a formatted message.
+
+    :param problem_sub_directory: The directory containing problems.
+    :type problem_sub_directory: str
+    :param result_filename: The name of the actual result file.
+    :type result_filename: str
+    :return: The lines which differ and a formatted message
+    :rtype: list of list of strings and str
```

Comment:
I think this may be clearer as a more concise description
```suggestion
    :rtype: list[list[str]], str
```